BM/tdx-compliance/tdx-compliance.c

```diff
@@ -475,6 +478,99 @@ const struct file_operations data_file_fops = {
 	.read = tdx_tests_proc_read,
 };
 
+static  unsigned long __lkm_order;
+static inline unsigned long lkm_read_cr0(void)
+{
+	unsigned long val;
+	asm volatile("mov %%cr0,%0\n\t" : "=r" (val), "=m" (__lkm_order));
+	return val;
+}
+static inline void lkm_write_cr0(unsigned long val)
+{
+	asm volatile("mov %0,%%cr0": : "r" (val), "m" (__lkm_order));
+}
+void disable_write_protection(void)
+{
+	unsigned long cr0 = lkm_read_cr0();
+	cr0 &= ~(1UL << 16);
+	lkm_write_cr0(cr0);
+}
+void enable_write_protection(void)
+{
+  unsigned long cr0 = lkm_read_cr0();
+  cr0 |= (1UL << 16);
+  lkm_write_cr0(cr0);
+}
+
+volatile static unsigned long interrupt_count=0;
+unsigned long orig_VE_handler_addr;
+// original #VE handler
+void (*orig_VE_handler)(void);
+
+// my #VE handler
+void my_VE_handler(void) 
+{
+	// interrupt_count++;
+	// pr_buf("#VE trigger\n");
+	// pr_info("#VE trigger\n");
+	// printk(KERN_DEBUG "#VE trigger\n");
+	return;
+}
+
+void my_VE_handler_asm_entry(void);
+__asm__(
+	".global my_VE_handler_asm_entry\n"
+	".type my_VE_handler_asm_entry, @function\n"
+	"my_VE_handler_asm_entry:\n"
+	// Save the registers
+	"pushq %rax\n\r"
+	"pushq %rcx\n\r"
+	"pushq %rdx\n\r"
+	"pushq %rdi\n\r"
+	"pushq %rsi\n\r"
+	"pushq %rbx\n\r"
+	"pushq %r8\n\r"
+	"pushq %r9\n\r"
+	"pushq %r10\n\r"
+	"pushq %r11\n\r"
+	"pushq %r12\n\r"
+	"pushq %r13\n\r"
+	"pushq %r14\n\r"
+	"pushq %r15\n\r"
```

Comment:
Will thing become proper if remove any instructions between the pushq r15 and popq r15 ? Of cause the jmp *orig_VE_handler_addr still necessary.